### PR TITLE
Aurora/bff datalayer

### PR DESCRIPTION
**Pull request by nsprenkle**
**TL;DR -** Split of functionality in mixins into data / UI components. Replaces #2044.

JIRA:
- https://2u-internal.atlassian.net/browse/AU-1398
- https://2u-internal.atlassian.net/browse/AU-1399
- https://2u-internal.atlassian.net/browse/AU-1400
- https://2u-internal.atlassian.net/browse/AU-1401
- https://2u-internal.atlassian.net/browse/AU-1402
- https://2u-internal.atlassian.net/browse/AU-1403
- https://2u-internal.atlassian.net/browse/AU-1404

**What changed?**

- Moved data gathering pieces of mixins into `xblock/apis`
- Created `xblock/ui_mixins/legacy` to replace existing view/handler behaviors from mixins, leveraging our new APIs.
- Moved remaining non-core-xblock functionality from `xblock` to `xblock/utils`.
- Updated references to changed file locations.

**Developer Checklist**

- [x] Reviewed the [release process](https://github.com/openedx/edx-ora2/blob/master/.github/release_process.md)
- [x] Translations and JS/SASS compiled
- [x] Bumped version number in [openassessment/\_\_init\_\_.py](https://github.com/openedx/edx-ora2/blob/master/openassessment/__init__.py#L4) and [package.json](https://github.com/openedx/edx-ora2/blob/master/package.json#L3)

**Testing Instructions**

- [x] Unit tests passing
- [x] Manual tests

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [x] I've tested the new functionality

FYI: @openedx/content-aurora

### CODE REVIEW

**Comment by nedbat** [comment]
Please check.

**Comment by nsprenkle** [comment]
Had tried opening and closing PRs to fix a build issue. Reverting to previous PR.

**Comment by nsprenkle** [comment]
JK, other PR can't be reopened because I did a squash

**Review by jansenk — APPROVED** [review]
